build.sc

```diff
@@ -573,7 +573,7 @@ object contrib extends MillModule {
       override def compileIvyDeps = T{
         Agg(
           // compile-time only, need to provide the correct scoverage version runtime
-          Deps.scalacScoveragePlugin,
+          Deps.scalacScoverageReporter,
```

Comment:
I didn't fully understand this. Is this _only_ being used to pull the version? I switched this to the reporter just to test it and it worked as well, but probably won't leave it as the reporter.

Reply:
The `contrib.scoverage.worker` is loaded by the `contrib.scoverage` module in a isolated classloader. The `contrib.scoverage.api` module is shared between both. So we can load the scoverage jars only when needed and isolated from other mill plugins. Technically, this allows us to use different scoverage versions in the same build. As Scoverage 2.x has an incompatible API, we need to create a new worker module working with Scoverage 2.x. We also do this in other plugin, e.g. `contrib.playlib`.